include/itkMorphologicalContourInterpolator.hxx

```
@@ -0,0 +1,193 @@
+                    ExpandRegion(resBB.first->second, ind);
+
+                std::pair<OrientationsType::iterator, bool> res
+                    = m_Orientations.insert(std::make_pair(val, ors));
+                OrientationsType::iterator oRef = res.first;
+                int cTrue = 0;
+                for (int a = 0; a < TImage::ImageDimension; a++)
+                {
+                    indPrev = ind;
+                    indPrev[a]--;
+                    indNext = ind;
+                    indNext[a]++;
+                    {
+                        if (input->GetPixel(indPrev) == 0 && input->GetPixel(indNext) == 0)
+                        {
+                            ors[a] = true;
+                            cTrue++;
+                        }
+                    }
+                }
+                if (cTrue == 1) //slice has empty adjacent space only along one axis
+                {
+                    for (int a = 0; a < TImage::ImageDimension; a++)
+                        oRef->second[a] = oRef->second[a] || ors[a]; //add this dimension for this label
+                }
+            }
+            ++it;
+        }
+
+        //widen bounding boxes by 1
+        region = output->GetRequestedRegion();
+        for (BoundingBoxesType::iterator it = m_BoundingBoxes.begin(); it != m_BoundingBoxes.end(); ++it)
+        {
+            for (int a = 0; a < TImage::ImageDimension; a++)
+            {
+                if (it->second.GetIndex(a) -1 <= region.GetIndex(a))
+                    it->second.SetIndex(a, it->second.GetIndex(a) - 1);
+                if (it->second.GetIndex(a) + it->second.GetSize(a) + 1 
+                    <= region.GetIndex(a) + region.GetSize(a))
+                    it->second.SetSize(a, it->second.GetSize(a) + 1);
+            }
+        }
+    }
+
+    template< class TImage>
+    void MorphologicalContourInterpolator<TImage>::
+        InterpolateAlong(int axis, typename TImage::Pointer out)
+    {
```

Maybe throw an exception here if it is TODO